### DISCUSSION TITLE
Why is the same unicode character displayed differently using the same font?

### DISCUSSION THREAD

**Discussion by ChenZhongPu** [kovidgoyal/kitty#6280] [Q&A] [answered]
I tested some private-user-area Unicode on several terminal apps on macOS using the same font (`FiraCode Nerd Font Mono
`), but they are displayed differently.

```sh
echo -e "\uF630"
```

In Kitty,

<img width="224" alt="image" src="https://github.com/kovidgoyal/kitty/assets/2310958/aebc81c8-a43b-4dca-bb4a-8411a52015c5">

In iTerm2,

<img width="134" alt="image" src="https://github.com/kovidgoyal/kitty/assets/2310958/80b45736-c044-443b-8475-cceb189bbaad">

In built-in Terminal,

<img width="152" alt="image" src="https://github.com/kovidgoyal/kitty/assets/2310958/3de610bb-9391-4501-94bf-65b194ae9c33">

As we can see, the display behavior of Kitty is kind of weird. Although PUA characters are not defined, I think they are related to specific fonts.

**Comment by kovidgoyal** [comment, marked as answer]
If you want to see what font kitty is using for any given character, use --debug-font-fallback. And if you want kitty to use a specific font for any unicode code point use symbol_map in kitty.conf

**Reply by ChenZhongPu** [reply]
Thanks. `symbol_map` works for me.  

```
symbol_map U+F630 MesloLGS NF
```

BTW, the result of `--debug-font-fallback` shows that I am indeed using `FiraCode Nerd Font Mono`:

```
❯ kitty --debug-font-fallback
[137 23:31:17.810320] Preloaded font faces:
[137 23:31:17.810345] normal face: /Users/zhongpu/Library/Fonts/FiraCodeNerdFontMono-Regular.ttf
[137 23:31:17.810351] bold face: /Users/zhongpu/Library/Fonts/FiraCodeNerdFontMono-SemiBold.ttf
[137 23:31:17.810355] italic face: /Users/zhongpu/Library/Fonts/FiraCodeNerdFontMono-Regular.ttf
[137 23:31:17.810359] bi face: /Users/zhongpu/Library/Fonts/FiraCodeNerdFontMono-Regular.ttf
```

But I still have no idea why this font cannot display `U+F630` in kitty.
